week2-6.6-decorator_factory.py

```diff
@@ -0,0 +1,44 @@
+from functools import wraps
+
+
+class MsgError(Exception):
+    """
+    a message error class for error messages if raised.
+    """
+    pass
+
+
+def type_check(correct_type):
+    """
+    this function is a decorator factory that checks if the type of argument given by user
+    is the one we want, if not it raises a custom error message to indicate that the type
+    added was wrong. if yes it returns the result to the wrapper, the wrapper returns it
+    to the decorator and the function returns the decorator.
+    :param correct_type: our correct type, in this case is INT
+    :return: returns a decorator
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            result = func(*args, **kwargs)
```

Comment:
You have a problem here.
You need to check the type of the arguments and not the type of the returned result.
Fix it and resubmit it order to improve your grade.